vivisect/parsers/pe.py

```diff
@@ -523,8 +523,12 @@ def loadPeIntoWorkspace(vw, pe, filename=None, baseaddr=None):
         pe.fd.seek(0)
         fbytes = pe.fd.read()
         for offset, i in pe_carve.carve(fbytes, 1):
-            # Found a sub-pe!
-            subpe = pe_carve.CarvedPE(fbytes, offset, [i])
+            try:
+                # ensure sub-pe can be parsed
+                subpe = pe_carve.CarvedPE(fbytes, offset, [i])
+            except vivisect.exc.CorruptPeFile:
```

Comment:
please insert an appropriate log message here, so an interested party can get an alert of the corrupt PE file.

thanks!

Reply:
addressed in 5cb6723